org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend

```diff
@@ -196,55 +203,42 @@ class DocumentProvider {
 		return cleanCreate(resourcePath, content, [push])
 	}
 	
 	@VisibleForTesting
-	def DocumentResource.createResult cleanCreate(String resourcePath, String content, Function<Void, ?> pushAction) {
-		var result = DocumentResource.createResult.badrequest
-		val preCommit = git.repository.resolve('HEAD')
-		logger.trace('commitid before any action is taken is ' + preCommit.getName)
-		try {
-			val file = workspaceProvider.getWorkspaceFile(resourcePath)
-			val created = workspaceProvider.create(file)
-			if (!created) {
-				result = DocumentResource.createResult.badrequest
-			} else {
-				if (!content.isNullOrEmpty) {
-					workspaceProvider.write(file, content)
-				}
-				val commitId = commit(file, '''add file: «file.name»''')
-				val pullResult = pull
-				val newCommitId = git.repository.resolve('HEAD')
-				if (newCommitId.equals(commitId) && pullResult.mergeResult.mergeStatus.successful) {
-					pushAction.apply(null)
-					result = DocumentResource.createResult.succeeded
+	def DocumentResource.ActionResult cleanCreate(String resourcePath, String content, Callable<?> pushAction) {
+		val file = workspaceProvider.getWorkspaceFile(resourcePath)
+		if (file.exists) {
+			logger.warn('''create failed, file «resourcePath» already exists''')
+			return DocumentResource.ActionResult.badrequest
+		} else {
+			return [|
+				val created = workspaceProvider.create(file)
+				if (!created) {
+					throw new RuntimeException('file could not be created')
 				} else {
-					if (!newCommitId.equals(commitId)) {
-						logger.warn('''unexpected inequality: pulled commit id '«newCommitId.getName»' != local commit id '«commitId.getName»' ''')
-						result = DocumentResource.createResult.repull
-					} else if (!pullResult.mergeResult.mergeStatus.successful) {
-						logger.warn('''merge conflicts in files «pullResult.mergeResult.conflicts.entrySet.map[key].toSet.join(', ')».''')
-						result = DocumentResource.createResult.repull
-					} else {
-						logger.warn('expected commit id not found or unknown merge conflict (unknown cause)')
-						result = DocumentResource.createResult.badrequest
+					if (!content.isNullOrEmpty) {
+						workspaceProvider.write(file, content)
 					}
+					return commit(file, '''add file: «file.name»''')
 				}
-			}
-		} catch (Exception e) {
-			logger.error('exception during save action (save and push)', e)
-			result = DocumentResource.createResult.repull
-		}
-		if (!result.equals(DocumentResource.createResult.succeeded)) {
-			logger.warn('resetting local repo to ' + preCommit.getName)
-			git.reset.setRef(preCommit.getName).setMode(ResetType.HARD).call
+			].wrapInCleanRepoActionWithCompensationOnFailure(pushAction, [
+				if (file.exists) {
+					// files not in the index, must be deleted, too (since resetting the index won't help)
+					logger.trace('''cleanup of remaining file '«file.absolutePath»' after reset hard necessary.''')
+					FileUtils.deleteQuietly(file)
+				} else {
```

Comment:
remove empty else block…?

Reply:
I had an eclipse warning if else block is missing